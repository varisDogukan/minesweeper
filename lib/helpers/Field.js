"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CellState = void 0;
var CellState = exports.CellState = {
  empty: 0,
  bomb: 9,
  hidden: 10,
  mark: 11,
  weakMark: 12
};