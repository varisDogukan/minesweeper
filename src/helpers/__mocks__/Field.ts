export const fieldGenerator = (size: number) => {
  switch (size) {
    case 9:
    default:
      return [
        [0, 0, 0, 0, 0, 0, 1, 9, 1],
        [0, 0, 0, 0, 1, 1, 2, 2, 2],
        [0, 0, 0, 0, 1, 9, 1, 1, 9],
        [0, 0, 1, 1, 3, 2, 2, 1, 1],
        [0, 0, 1, 9, 2, 9, 1, 0, 0],
        [1, 1, 2, 1, 2, 1, 1, 1, 1],
        [1, 9, 1, 0, 0, 0, 0, 1, 9],
        [1, 1, 1, 1, 1, 2, 1, 3, 2],
        [0, 0, 0, 1, 9, 2, 9, 2, 9],
      ];

    case 16:
      return [
        [9, 2, 2, 9, 2, 9, 1, 0, 0, 1, 9, 1, 0, 0, 0, 0],
        [4, 9, 3, 2, 3, 2, 1, 0, 0, 2, 2, 2, 0, 1, 1, 1],
        [9, 9, 12, 1, 9, 1, 0, 0, 0, 1, 9, 1, 0, 1, 9, 2],
        [2, 2, 1, 1, 1, 2, 2, 3, 2, 2, 1, 1, 0, 1, 2, 9],
        [0, 0, 0, 0, 0, 1, 9, 9, 9, 1, 0, 1, 1, 1, 1, 1],
        [0, 0, 0, 0, 0, 1, 2, 4, 3, 3, 1, 2, 9, 2, 1, 0],
        [0, 0, 0, 0, 1, 1, 1, 1, 9, 2, 9, 2, 2, 9, 1, 0],
        [0, 1, 1, 1, 1, 9, 1, 1, 1, 2, 1, 2, 2, 2, 2, 1],
        [0, 1, 9, 1, 1, 1, 1, 1, 2, 2, 1, 1, 9, 1, 2, 9],
        [1, 2, 2, 1, 1, 1, 2, 2, 9, 9, 1, 1, 1, 1, 3, 9],
        [1, 9, 1, 0, 1, 9, 3, 9, 4, 2, 1, 0, 0, 1, 3, 9],
        [1, 1, 1, 0, 1, 1, 3, 9, 2, 1, 1, 1, 0, 1, 9, 2],
        [0, 1, 1, 1, 1, 1, 3, 3, 3, 2, 9, 1, 0, 2, 2, 2],
        [0, 1, 9, 1, 2, 9, 4, 9, 9, 4, 3, 2, 0, 1, 9, 1],
        [0, 1, 1, 2, 3, 9, 9, 3, 3, 9, 9, 3, 1, 2, 1, 1],
        [0, 0, 0, 1, 9, 3, 2, 1, 1, 3, 9, 3, 9, 1, 0, 0],
      ];

    case 22:
      return [
        [1, 9, 3, 9, 2, 1, 1, 2, 2, 3, 9, 1, 1, 2, 9, 2, 9, 9, 2, 9, 3, 9],
        [2, 2, 4, 9, 3, 3, 9, 4, 9, 9, 2, 1, 1, 9, 2, 2, 3, 3, 3, 2, 4, 9],
        [1, 9, 3, 2, 2, 9, 9, 5, 9, 3, 1, 0, 2, 3, 3, 2, 3, 9, 2, 1, 9, 2],
        [3, 4, 9, 1, 1, 3, 9, 3, 1, 2, 1, 2, 3, 9, 9, 2, 9, 9, 2, 2, 2, 2],
        [9, 9, 3, 1, 0, 2, 2, 2, 0, 1, 9, 3, 9, 9, 3, 2, 2, 3, 2, 2, 9, 1],
        [4, 9, 3, 1, 0, 1, 9, 1, 0, 1, 3, 9, 4, 2, 1, 0, 0, 1, 9, 3, 2, 1],
        [9, 3, 9, 2, 1, 3, 2, 3, 1, 1, 3, 9, 3, 0, 1, 1, 1, 1, 3, 9, 2, 0],
        [2, 3, 2, 2, 9, 3, 9, 3, 9, 1, 2, 9, 3, 1, 2, 9, 1, 0, 2, 9, 2, 0],
        [1, 9, 1, 2, 2, 4, 9, 4, 2, 2, 1, 1, 3, 9, 3, 1, 1, 0, 1, 2, 3, 2],
        [1, 1, 1, 1, 9, 3, 2, 3, 9, 1, 1, 1, 3, 9, 2, 0, 0, 0, 0, 1, 9, 9],
        [0, 0, 0, 1, 1, 2, 9, 2, 2, 2, 2, 9, 2, 1, 2, 1, 1, 0, 0, 1, 2, 2],
        [0, 0, 1, 1, 1, 1, 1, 1, 1, 9, 3, 2, 2, 1, 2, 9, 2, 2, 2, 1, 0, 0],
        [0, 1, 2, 9, 1, 0, 0, 1, 2, 3, 9, 1, 1, 9, 2, 1, 2, 9, 9, 1, 0, 0],
        [0, 1, 9, 3, 3, 2, 1, 1, 9, 3, 2, 1, 1, 1, 1, 0, 1, 2, 2, 1, 0, 0],
        [0, 1, 1, 2, 9, 9, 2, 2, 4, 9, 2, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 0],
        [0, 0, 0, 2, 3, 4, 3, 9, 5, 9, 3, 2, 9, 2, 0, 0, 0, 0, 1, 9, 2, 1],
        [1, 1, 0, 1, 9, 3, 9, 4, 9, 9, 3, 2, 9, 2, 0, 1, 1, 1, 1, 2, 9, 1],
        [9, 2, 0, 2, 3, 5, 9, 3, 3, 9, 2, 1, 1, 1, 0, 2, 9, 3, 1, 2, 2, 2],
        [9, 3, 1, 1, 9, 9, 2, 1, 1, 2, 2, 1, 0, 0, 0, 3, 9, 5, 9, 2, 2, 9],
        [3, 9, 2, 1, 2, 2, 1, 0, 0, 1, 9, 1, 0, 0, 0, 2, 9, 4, 9, 2, 2, 9],
        [2, 9, 3, 1, 0, 1, 1, 2, 1, 2, 1, 2, 1, 1, 0, 1, 1, 2, 2, 2, 2, 1],
        [1, 2, 9, 1, 0, 1, 9, 2, 9, 1, 0, 1, 9, 1, 0, 0, 0, 0, 1, 9, 1, 0],
      ];
  }
};

export const CellState = {
  empty: 0,
  bomb: 9,
  hidden: 10,
  flag: 11,
  weakFlag: 12,
};

export function generateFieldWithDefaultState(
  size: number,
  state = CellState.empty
) {
  return new Array(size).fill(null).map(() => new Array(size).fill(state));
}
