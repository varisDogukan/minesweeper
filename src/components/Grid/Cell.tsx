import React from "react";
import styled from "@emotion/styled";

import { Cell as CellType, CellState, Coords } from "@/helpers/Field";
import { useMouseDown } from "@/hooks/useMouseDown";

export interface CellProps {
  children: CellType;
  coords: Coords;
  onClick: (coords: Coords) => void;
  onContextMenu: (coords: Coords) => void;
}

export function isActiveCell(cell: CellType): boolean {
  return [CellState.hidden, CellState.flag, CellState.weakFlag].includes(cell);
}

export default function Cell({ children, coords, ...rest }: CellProps) {
  const [mouseDown, onMouseDown, onMouseUp] = useMouseDown();

  const onClick = () => {
    rest.onClick(coords);
  };

  const onContextMenu = (elem: React.MouseEvent<HTMLElement>) => {
    elem.preventDefault();

    if (isActiveCell(children)) {
      rest.onContextMenu(coords);
    }
  };

  const props = {
    onClick,
    onContextMenu,
    onMouseDown,
    onMouseUp,
    onMouseLeave: onMouseUp,
    mouseDown,
    "data-testid": `${children}_${coords}`,
  };

  return <ComponentsMap {...props}>{children}</ComponentsMap>;
}

interface ComponentsMapProp {
  children: CellType;
  onClick: (elem: React.MouseEvent<HTMLElement>) => void;
  onContextMenu: (elem: React.MouseEvent<HTMLElement>) => void;
  onMouseDown: () => void;
  onMouseUp: () => void;
  onMouseLeave: () => void;
  mouseDown: boolean;
  "data-testid"?: string;
}

function ComponentsMap({ children, ...rest }: ComponentsMapProp) {
  const nonActiveCellProps = {
    onContextMenu: rest.onContextMenu,
    "data-testid": rest["data-testid"],
  };

  switch (children) {
    case CellState.empty:
      return <RevealedFrame {...nonActiveCellProps} />;

    case CellState.bomb:
      return (
        <BombFrame {...nonActiveCellProps}>
          <Bomb />
        </BombFrame>
      );

    case CellState.flag:
      return (
        <ClosedFrame {...rest}>
          <Flag />
        </ClosedFrame>
      );

    case CellState.weakFlag:
      return (
        <ClosedFrame {...rest}>
          <WeakFlag />
        </ClosedFrame>
      );

    case CellState.hidden:
      return <ClosedFrame {...rest} />;

    default:
      return <RevealedFrame {...nonActiveCellProps}>{children}</RevealedFrame>;
  }
}

interface ClosedFrameProps {
  mouseDown?: boolean;
}

export const ClosedFrame = styled.div<ClosedFrameProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  user-select: none;
  cursor: pointer;
  width: 1.8vw;
  height: 1.8vw;
  background-color: #d1d1d1;
  border: 0.6vh solid transparent;
  border-color: ${({ mouseDown = false }) =>
    mouseDown ? "transparent" : "white #9e9e9e #9e9e9e white"};
  transition: all 0.2s linear;

  &:hover {
    filter: brightness(1.1);
  }
`;

const transparent = "rgba(0,0,0,0)";
const colors: { [key in CellType]: string } = {
  0: "#000",
  1: "#2a48ec",
  2: "#2bb13d",
  3: "#ec6561",
  4: "#23db7",
  5: "#a6070f",
  6: "#e400af",
  7: "#906a02",
  8: "#fa0707",
  9: transparent,
  10: transparent,
  11: transparent,
  12: transparent,
};

const RevealedFrame = styled(ClosedFrame)`
  border-color: #ddd;
  cursor: default;
  color: ${({ children }) => colors[children as CellType] ?? transparent};

  &:hover {
    filter: brightness(1.1);
  }
`;

const Bomb = styled.div`
  border-radius: 50%;
  width: 1vw;
  height: 1vw;
  background-color: #333;
`;

const BombFrame = styled(RevealedFrame)`
  background-color: #ec433c;
`;

const Flag = styled.div`
  width: 0px;
  height: 0pxw;
  border-top: 0.5vw solid transparent;
  border-bottom: 0.5vw solid transparent;
  border-left: 0.5vw solid #ec433c;
`;

const WeakFlag = styled(Flag)`
  border-left: 0.5vw solid #f19996;
`;
