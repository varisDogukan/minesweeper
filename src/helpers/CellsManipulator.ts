import { Cell, CellState, Coords, Field } from "./Field";

export function getNeigboursItems([y, x]: Coords): Record<
  string,
  [number, number]
> {
  return {
    top: [y - 1, x],
    topRight: [y - 1, x + 1],
    right: [y, x + 1],
    rightBottom: [y + 1, x + 1],
    bottom: [y + 1, x],
    bottomLeft: [y + 1, x - 1],
    left: [y, x - 1],
    leftTop: [y - 1, x - 1],
  };
}

export function checkItemInField([y, x]: Coords, { length }: Field): boolean {
  return y >= 0 && x >= 0 && length - y > 0 && length - x > 0;
}

export function incrementNeibours(coords: Coords, field: Field): Field {
  const items = getNeigboursItems(coords);

  for (const item of Object.values(items)) {
    if (checkItemInField(item, field)) {
      const [y, x] = item;
      const cell = field[y][x];

      if (cell < 8) {
        field[y][x] = (cell + 1) as Cell;
      }
    }
  }
  return field;
}

/**
 * Open cell in the player field using game field info
 * @param {Coords} coords
 * @param {Field} playerField
 * @param {Field} gameField
 * @returns {Field}
 */
export function openCell(
  coords: Coords,
  playerField: Field,
  gameField: Field
): Field {
  const [y, x] = coords;
  const gameCell = gameField[y][x];

  if (gameCell === CellState.bomb) {
    throw new Error("Game Over");
  }

  if (gameCell === CellState.empty) {
    playerField[y][x] = gameCell;

    const items = getNeigboursItems(coords);

    for (const coords of Object.values(items)) {
      if (checkItemInField(coords, gameField)) {
        const [y, x] = coords;

        const gameCell = gameField[y][x];
        const playerCell = playerField[y][x];

        if (gameCell === CellState.hidden && playerCell !== CellState.bomb) {
          playerField = openCell(coords, playerField, gameField);
        }

        if (gameCell < CellState.bomb) {
          playerField[y][x] = gameCell;
        }
      }
    }
  }

  playerField[y][x] = gameCell;

  return playerField;
}
