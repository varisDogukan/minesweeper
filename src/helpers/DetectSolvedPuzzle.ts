import { CellState, Field } from "./Field";

/**
 * Detect solved puzzle based on the player and game fields coorelation
 * @param {Field} playerField
 * @param {Field} gameField
 * @returns  {[boolean, number]}
 */
export default function detectSolvedPuzzle(
  playerField: Field,
  gameField: Field
): [boolean, number] {
  const { hidden, bomb, flag, weakFlag } = CellState;

  let bombsCounter = 0;
  let flagCounter = 0;
  let detectedBombsCounter = 0;
  let hiddenCounter = 0;

  for (const y of gameField.keys()) {
    for (const x of gameField[y].keys()) {
      const gameCell = gameField[y][x];
      const playerCell = playerField[y][x];

      if (playerCell === hidden) hiddenCounter++;

      if ([flag, weakFlag].includes(playerCell)) flagCounter++;

      if (gameCell === bomb) {
        bombsCounter++;

        if (playerCell === flag) detectedBombsCounter++;
      }
    }
  }

  const isPuzzleSolved =
    bombsCounter === detectedBombsCounter && hiddenCounter === 0;

  return [isPuzzleSolved, flagCounter];
}
