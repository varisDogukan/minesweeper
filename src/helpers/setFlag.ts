import { CellState, Coords, Field } from "./Field";

/**
 * Set flag to the cell
 * @param {Coords} coords
 * @param {Field} playerField
 * @param {Field} gameField
 * @returns {[Field, FlagCounter]}
 */
export default function setFlag(
  coords: Coords,
  playerField: Field,
  gameField: Field
): Field {
  const [y, x] = coords;
  const cell = playerField[y][x];

  const { flag, weakFlag, hidden } = CellState;

  switch (cell) {
    case flag:
      playerField[y][x] = weakFlag;
      break;

    case hidden:
      playerField[y][x] = flag;
      break;

    case weakFlag:
      playerField[y][x] = hidden;
      break;
  }

  return playerField;
}
