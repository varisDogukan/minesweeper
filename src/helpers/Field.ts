import { incrementNeibours } from "./CellsManipulator";

export type Cell = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
export type Field = Array<Cell[]>;
export type Coords = [number, number];

export const CellState: Record<string, Cell> = {
  empty: 0,
  bomb: 9,
  hidden: 10,
  flag: 11,
  weakFlag: 12,
};

/**
 * Create empty field
 *
 * Example emptyFieldGenerator(2)
 * [
 *  [0, 0],
 *  [0, 0]
 * ]
 *
 * @param {number} size
 * @param {Cell} state
 * @returns {Field}
 */
export function emptyFieldGenerator(
  size: number,
  state: Cell = CellState.empty
): Field {
  return new Array(size).fill(null).map(() => new Array(size).fill(state));
}

export function fieldGenerator(size: number, probability: number): Field {
  if (probability < 0 || probability > 1) {
    throw new Error("Probability must be between 0 and 1");
  }

  let unprocessedCells = size * size;
  let restCellsWithBombs = unprocessedCells * probability;

  const result: Field = emptyFieldGenerator(size);

  for (let y = 0; y < size; y++) {
    for (let x = 0; x < size; x++) {
      if (restCellsWithBombs / unprocessedCells > Math.random()) {
        result[y][x] = CellState.bomb;

        incrementNeibours([y, x], result);

        restCellsWithBombs--;
      }

      unprocessedCells--;
    }
  }

  return result;
}
