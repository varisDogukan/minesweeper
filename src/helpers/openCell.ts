import { checkItemInField, getNeigboursItems } from "./CellsManipulator";
import { CellState, Coords, Field } from "./Field";

/**
 * Open cell in the player field using game field info
 * @param {Coords} coords
 * @param {Field} playerField
 * @param {Field} gameField
 * @returns {Field}
 */
export default function openCell(
  coords: Coords,
  playerField: Field,
  gameField: Field
): Field {
  const [y, x] = coords;
  const gameCell = gameField[y][x];

  if (gameCell === CellState.bomb) {
    throw new Error("Game Over");
  }

  if (gameCell === CellState.empty) {
    playerField[y][x] = gameCell;

    const items = getNeigboursItems(coords);

    for (const coords of Object.values(items)) {
      if (checkItemInField(coords, gameField)) {
        const [y, x] = coords;

        const gameCell = gameField[y][x];
        const playerCell = playerField[y][x];

        if (gameCell === CellState.hidden && playerCell !== CellState.bomb) {
          playerField = openCell(coords, playerField, gameField);
        }

        if (gameCell < CellState.bomb) {
          playerField[y][x] = gameCell;
        }
      }
    }
  }

  playerField[y][x] = gameCell;

  return playerField;
}
