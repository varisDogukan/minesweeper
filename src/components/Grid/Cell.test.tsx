import React from "react";
import { render, screen, fireEvent, createEvent } from "@testing-library/react";

import { CellState, Coords } from "@/helpers/Field";

import Cell, { isActiveCell, ClosedFrame } from "./Cell";

beforeEach(() => {
  jest.clearAllMocks();
});

describe("Cell component check", () => {
  const coords: Coords = [1, 1];

  for (let cell = CellState.empty; cell <= CellState.weakFlag; cell++) {
    it("Cell renders correctly", () => {
      const props = {
        coords,
        onClick: jest.fn(),
        onContextMenu: jest.fn(),
      };

      const { asFragment } = render(<Cell {...props}>{cell}</Cell>);

      expect(asFragment()).toMatchSnapshot();
    });

    it("Cell Frame renders correctly", () => {
      const { asFragment } = render(<ClosedFrame mouseDown={true} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("Check prevent default contextMenu for every type of cell", () => {
      const props = {
        coords,
        onClick: jest.fn(),
        onContextMenu: jest.fn(),
      };

      render(<Cell {...props}>{cell}</Cell>);

      const cellComp = screen.getByTestId(`${coords}`);
      const contextMenuEvent = createEvent.contextMenu(cellComp);
      fireEvent(cellComp, contextMenuEvent);

      expect(contextMenuEvent.defaultPrevented).toBe(true);
    });

    it("onClick and onContextMenu handler should be called for active cells", () => {
      const props = {
        coords,
        onClick: jest.fn(),
        onContextMenu: jest.fn(),
      };

      render(<Cell {...props}>{cell}</Cell>);

      const cellComp = screen.getByTestId(`${coords}`);
      fireEvent.click(cellComp);
      fireEvent.contextMenu(cellComp);

      if (isActiveCell(cell)) {
        expect(props.onClick).toHaveBeenCalled();
        expect(props.onContextMenu).toHaveBeenCalled();
      } else {
        expect(props.onClick).not.toHaveBeenCalled();
        expect(props.onContextMenu).not.toHaveBeenCalled();
      }
    });
  }
});
