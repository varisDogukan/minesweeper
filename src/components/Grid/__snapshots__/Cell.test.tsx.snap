// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component check Cell Frame renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 2`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 3`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 4`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 5`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 6`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 7`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 8`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 9`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 10`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 11`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 12`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell Frame renders correctly 13`] = `
<DocumentFragment>
  <div
    class="css-1q0n4ib"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-103121h"
    data-testid="1,1"
    role="cell"
  >
    0
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 2`] = `
<DocumentFragment>
  <div
    class="css-1dokkwi"
    data-testid="1,1"
    role="cell"
  >
    1
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 3`] = `
<DocumentFragment>
  <div
    class="css-hxpcqh"
    data-testid="1,1"
    role="cell"
  >
    2
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 4`] = `
<DocumentFragment>
  <div
    class="css-x1wks9"
    data-testid="1,1"
    role="cell"
  >
    3
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 5`] = `
<DocumentFragment>
  <div
    class="css-10uw94w"
    data-testid="1,1"
    role="cell"
  >
    4
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 6`] = `
<DocumentFragment>
  <div
    class="css-1snmbr7"
    data-testid="1,1"
    role="cell"
  >
    5
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 7`] = `
<DocumentFragment>
  <div
    class="css-183elvv"
    data-testid="1,1"
    role="cell"
  >
    6
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 8`] = `
<DocumentFragment>
  <div
    class="css-1okt1ig"
    data-testid="1,1"
    role="cell"
  >
    7
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 9`] = `
<DocumentFragment>
  <div
    class="css-1bhuacv"
    data-testid="1,1"
    role="cell"
  >
    8
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 10`] = `
<DocumentFragment>
  <div
    class="css-16bwjvk"
    data-testid="1,1"
    role="cell"
  >
    <div
      class="css-1f7ob6g"
    />
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 11`] = `
<DocumentFragment>
  <div
    class="css-1yh9r0j"
    data-testid="1,1"
    role="cell"
  >
    10
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 12`] = `
<DocumentFragment>
  <div
    class="css-1yh9r0j"
    data-testid="1,1"
    role="cell"
  >
    <div
      class="css-1duh8q7"
    />
  </div>
</DocumentFragment>
`;

exports[`Cell component check Cell renders correctly 13`] = `
<DocumentFragment>
  <div
    class="css-1yh9r0j"
    data-testid="1,1"
    role="cell"
  >
    <div
      class="css-11asmo2"
    />
  </div>
</DocumentFragment>
`;
